dev/validate_bom__config.py

```diff
@@ -899,6 +913,179 @@ def add_files_to_upload(self, options, file_set):
+                               options.gcs_pubsub_project,
+                               options.gcs_pubsub_credentials_path,
+                               options.gcs_pubsub_subscription))
+    if options.gcs_pubsub_subscription != options.pubsub_google_subscription_name:
+      raise ValueError('Inconsistent pub/sub subscription reference in configuration: '
+                       ' --gcs_pubsub_subscription="{}"'
+                       ' --pubsub_google_subscription_name="{}"'
+                       .format(options.gcs_pubsub_subscription,
+                               options.pubsub_google_subscription_name))
+
+  def __instantiate_clients(self, options):
+    """Instantiates and returns publisher, subscriber, and storage clients.
+
+    Returns:
+      Tuple of (publisher_client, subscriber_client, storage_client)
+    """
+    SCOPES = [
+      'https://www.googleapis.com/auth/devstorage.full_control',
+      'https://www.googleapis.com/auth/cloud-platform',
+      'https://www.googleapis.com/auth/pubsub',
+    ]
+    credentials = service_account.Credentials.from_service_account_file(
+      options.gcs_pubsub_credentials_path,
+      scopes=SCOPES)
+    publisher_client = pubsub.PublisherClient(credentials=credentials)
+    subscriber_client = pubsub.SubscriberClient(credentials=credentials)
+    storage_client = storage.Client(credentials=credentials, project=options.gcs_pubsub_project)
+    return (publisher_client, subscriber_client, storage_client)
+
+  def setup_environment(self, options):
+    """Implements interface."""
+    publisher_client, subscriber_client, storage_client = self.__instantiate_clients(options)
+
+    logging.info('Creating topic %s in project %s', options.gcs_pubsub_topic, options.gcs_pubsub_project)
+    topic_ref = publisher_client.topic_path(options.gcs_pubsub_project, options.gcs_pubsub_topic)
+    publisher_client.create_topic(topic_ref)
+
+    logging.info('Creating subscription %s in project %s', options.gcs_pubsub_subscription, options.gcs_pubsub_project)
+    subscription_ref = subscriber_client.subscription_path(options.gcs_pubsub_project,
+                                                           options.gcs_pubsub_subscription)
+    subscriber_client.create_subscription(subscription_ref, topic_ref)
+    storage_client.create_bucket(options.gcs_pubsub_bucket)
+
+    bucket = storage_client.get_bucket(options.gcs_pubsub_bucket)
+    notification = bucket.notification(options.gcs_pubsub_topic, topic_project=options.gcs_pubsub_project,
+                                       payload_format=storage.notification.JSON_API_V1_PAYLOAD_FORMAT)
+    notification.create()
+    logging.debug('Created bucket notification %s', notification)
+
+  def teardown_environment(self, options):
+    """Implements interface."""
+    publisher_client, subscriber_client, storage_client = self.__instantiate_clients(options)
+
+    build_number = os.environ['BUILD_NUMBER']
```

Comment:
why BUILD_NUMBER? I dont see this used elsewhere. I'd prefer it be a parameter if you need one rather than environment variable because this should be easily run by people in shells, not just a CI server.

Reply:
Unused, forgot to remove in a refactor.